File: fs/BeginnersSelection/07_KagamiMochi.fs
// https://atcoder.jp/contests/abs/tasks/abc085_b

stdin.ReadLine()
|> int
|> fun x -> [1..x]
|> List.map (fun _ -> stdin.ReadLine())
|> List.distinct
|> List.length
|> printfn "%d"

// ----------

stdin.ReadLine() |> int |> fun x -> [1..x] |> List.map (fun _ -> stdin.ReadLine()) |> List.distinct |> List.length |> printfn "%d" 

// https://atcoder.jp/contests/abs/submissions/9414793

File: ex/BeginnersSelection/07_KagamiMochi.ex
# https://atcoder.jp/contests/abs/tasks/abc085_b

defmodule Main, do:
  def main(), do:
    IO.read(:line)
    |> String.trim
    |> String.to_integer
    |> (& (1..&1)).()
    |> Enum.map(fn _ -> IO.read(:line) end)
    |> MapSet.new
    |> MapSet.size
    |> IO.puts

# https://atcoder.jp/contests/abs/submissions/21759385
